test/preparser.js

```diff
@@ -29,20 +30,68 @@ describe('preparser', function () {
   });
 
   describe('environmental variables', function () {
-    it('should convert them according to os', function () {
-      if (windows) {
-        cash('echo %PATH%').should.equal(`${path}\n`);
-      } else {
-        cash('echo $PATH').should.equal(`${path}\n`);
+    before(function () {
+      process.env.FOO = 'This*string\'has   $pecial${characters}';
+    });
+    it('should convert simple variable references', function () {
+      cash('echo $PATH').should.equal(`${path}\n`);
+    });
+
+    it('should convert variable references using the ${.*} syntax', function () {
+      cash('echo ${PATH}').should.equal(`${path}\n`);
+    });
+
+    it('should concatenate variables with no space', function () {
+      cash('echo $PATH$PATH').should.equal(`${path}${path}\n`);
+      cash('echo ${PATH}${PATH}').should.equal(`${path}${path}\n`);
+    });
+
+    it('should display variables with one or more spaces in between', function () {
+      cash('echo $PATH $PATH').should.equal(`${path} ${path}\n`);
+      cash('echo ${PATH} ${PATH}').should.equal(`${path} ${path}\n`);
+    });
+
+    it('should display variable references from inside double quotes', function () {
+      cash('echo "$PATH   $PATH"').should.equal(`${path}   ${path}\n`);
+    });
+
+    it('should separate variable references from strings with slash or dot', function () {
+      cash('echo foo/$PATH/$PATH.bar').should.equal(`foo/${path}/${path}.bar\n`);
+    });
+
+    it('should show non-existent variables as the empty string', function () {
+      cash('echo $FAKE_ENV_VAR').should.equal('\n');
+    });
+
+    it('should allow underscores in environmental variable names', function () {
+      if (nvmDir) {
+        cash('echo $NVM_DIR').should.equal(`${nvmDir}\n`);
       }
     });
 
-    it.skip('should convert the same variable twice', function () {
+    it('should append variables and strings', function () {
+      cash('echo foo${PATH}bar').should.equal(`foo${path}bar\n`);
+    });
+
+    it('should have proper case sensitivity', function () {
       if (windows) {
-        cash('echo %PATH%-%PATH%').should.equal(`${path}-${path}\n`);
+        // Case insensitive
```

Comment:
Actually, the issue here is that I put `%path%` instead of `$path` (I forgot to change it). Fixed this, and I'll push again.